### ISSUE TITLE
wp_remote_post empty or timed out

### ISSUE THREAD

**Issue by dbeja** [A5hleyRich/wp-background-processing#35]
Hi there,

I'm trying to use wp-background-processing, but it never enters task function (or even maybe_handle).
I debugged wp_remote_post but the response is always empty, and if I changed blocked to true on post parameters I get a timeout.

Any help/ideas debugging this?

Thanks

**Comment by dbeja** [comment]
Found the problem!

The constructor of WP_Async_Request adds the ajax actions so these need to be called at the right time and I was initializing my class at the callback of another action, at a point where these ajax actions couldn't be initialized anymore.

I'm using wp-background-processing on a theme, so what I did was to create a global variable $async_request and initialize it on init:

    add_action( 'init', 'init_my_theme' );
    function init_my_theme() {
    	global $async_request;
    	require_once plugin_dir_path( __FILE__ ) . 'wp-background-processing/wp-background-processing.php';
    	require_once plugin_dir_path( __FILE__ ) . 'async-request.php';
    
    	$async_request = new Async_Request();
    }

Then, whenever I need to use it I just call:

    global $async_request;
    $async_request->dispatch();

**Comment by dskvr** [comment]
I've been trying to figure out for a while how this is a solution, as I am incredibly desperate to find one. `require_once plugin_dir_path( __FILE__ ) . 'wp-background-processing/wp-background-processing.php';` is already included by the plugin in `/wp-background-processing/wp-background-processing.php` like so ```require_once plugin_dir_path( __FILE__ ) . 'classes/wp-async-request.php';
require_once plugin_dir_path( __FILE__ ) . 'classes/wp-background-process.php';```

Can you describe how this works? Or is this only relevant for your very particular and specific implementation?

**Comment by dbeja** [comment]
Sorry, I don't know if I understand your question...

But I'm not using the sample plugin.
What I did was add the wp-background-processing directory inside my theme/plugin, then inside my theme/plugin initializations I import the file wp-background-processing/wp-background-processing.php which it also imports the needed classes.
And I also import the class I created for my specific case (async-request.php in my case).
